wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -1085,10 +1087,91 @@ private class WasmExpressionBuilder private (
             genIsPrimType(primType)
           case None =>
             val info = ctx.getClassInfo(testClassName)
+
+            // TODO: Efficient type inclusion test
+            // Current implementation walk through the itables of the Object which takes O(N)
+            // (where N = number of interfaces the expr implements)
+            // Also, this procedure can't be extracted as a helper function in Wasm because the
+            // immediate argument of `ref.test` is specific to the `testClassName`,
+            // which result in a number of instructions everywhere at `isInstanceOf[interface]`.
```

Comment:
Have you considered making one function *per* target `testClassName`? That would be similar to the `newDefault` functions, which we have one per class. You can even know in advance which ones you need by testing `linkedClass.hasInstanceTests`.

Reply:
> You can even know in advance which ones you need by testing linkedClass.hasInstanceTests.

Sounds nice :+1: will do that

Reply:
done https://github.com/tanishiking/scala-wasm/pull/29/commits/56f36852168a63b92c57dfcfae9feb05d1513dec 😄